File: 11. Container With Most Water/solution.py
class Solution:
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        if len(height) < 2:
            return 0
        
        n = len(height)
        area = 0
        i = 0
        j = n - 1
        
        while i < j:
            area = max((j-i)*min([height[i], height[j]]), area)
            if height[i] > height[j]:
                j -= 1
            else:
                i += 1
                
        return area
            
        

File: 11. Container With Most Water/solution.java
class Solution {
    public int maxArea(int[] height) {
        int area = 0, maxArea = 0;
        int i = 0;
        int j = height.length-1;
        while (i < j) {
            area = Math.min(height[j], height[i])*(j-i);
            if (area > maxArea) {
                maxArea = area;
            }
            if (height[i] > height[j]) {
                j--;
            }
            else {
                i++;
            }
        }
        return maxArea;
    }
}
